core/src/main/java/hudson/util/io/TarArchiver.java

```diff
@@ -73,6 +73,7 @@ public void visitSymlink(File link, String target, String relativePath) throws I
 
         tar.putArchiveEntry(e);
         tar.closeArchiveEntry();
+        consumer.accept(relativePath);
```

Comment:
shouldn't we enforce nothing wrong happens on the consumer? like try-catching it to avoid bad clients making the archiver to fail.